commons-shared/src/main/scala/com/avsystem/commons/SharedExtensions.scala

```diff
@@ -345,11 +340,26 @@ object SharedExtensions extends SharedExtensions {
     def containsAll(other: BTraversable[A]): Boolean = other.forall(set.contains)
   }
 
+  class TraversableOps[C[X] <: BTraversable[X], A](private val coll: C[A]) extends AnyVal {
+
+    def headOpt: Opt[A] = if (coll.isEmpty) Opt.Empty else Opt(coll.head)
+
+    def lastOpt: Opt[A] = if (coll.isEmpty) Opt.Empty else Opt(coll.last)
+
+    def mkMap[K, V](keyFun: A => K, valueFun: A => V): Map[K, V] = {
```

Comment:
This could also be in `TraversableOnceOps` if you use `foreach` instead of `map` (like it was done before).